### PR TITLE
#151168605 Refactor get notifications

### PR DESCRIPTION
**Pull request by Mnickii**
#### What does this PR do?
It implements get all notifications for a logged in user.
#### Description of Task to be completed?

- create endpoint `GET api/v1/notifications/:notif_id` to get a single notification
- create enpoint `GET api/v1/notifications` to get all notifications of a user
- update tests 
#### How should this be manually tested?
`pytest -v` 
#### What are the relevant pivotal tracker stories?
 #151168605

### CODE REVIEW

File: api/views/notifications.py [hunk 1]
```diff
@@ -14,12 +14,15 @@ class NotificationsResource(Resource):
     def get(self, notif_id=None):
         user_id = g.current_user.user_id
         if notif_id:
-            notifications = Notification.query_redis(user_id, notif_id)
-            response = {'status': 'sucess', 'data': notifications}
+            try:
+                notifications = Notification.query_redis(user_id, notif_id)
+                response = {'status': 'success', 'data': notifications}
+            except KeyError:
```
Comment (andela-ojoloko) [inline api/views/notifications.py]: can't this be done in the models ??

Comment (Mnickii) [reply api/views/notifications.py]: This has been handled in the models, but it returns an error 500. I needed to catch it to give a 404 and a message.

Comment (georgreen) [reply api/views/notifications.py]: @andela-ojoloko The model throws exceptions when it receives unexpected input, this enables the person calling the function to be in control as they can handle the exception and choose an action that can handle's the error.

**Comment by andela-ojoloko** [comment]
LGTM

**Pull request merged**